zh-learn-application/src/main/java/com/zhlearn/application/service/ParallelWordAnalysisService.java

```diff
@@ -19,13 +20,27 @@ public class ParallelWordAnalysisService implements WordAnalysisService {
     private final WordAnalysisServiceImpl delegate;
     private final ExecutorService executorService;
 
-    public ParallelWordAnalysisService(ProviderRegistry registry, int threadPoolSize) {
-        this.delegate = new WordAnalysisServiceImpl(registry);
+    public ParallelWordAnalysisService(ExampleProvider exampleProvider,
+                                     ExplanationProvider explanationProvider,
+                                     StructuralDecompositionProvider decompositionProvider,
+                                     PinyinProvider pinyinProvider,
+                                     DefinitionProvider definitionProvider,
+                                     AudioProvider audioProvider,
+                                     int threadPoolSize) {
+        this.delegate = new WordAnalysisServiceImpl(exampleProvider, explanationProvider,
```

Comment:
wouldn't it be cleaner to invert the logic here? I.e, create the WordAnalysisServiceImpl outside and pass it in to the constructor?